app/controllers/graphic_requests_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+class GraphicRequestsController < ApplicationController
+
+  def show
+    graphic_request = GraphicRequest.where(download_hash: params[:id]).first
```

Comment:
maybe handle not found here? in case people copy/paste their links and forget characters :)